smt/utils/test/test_design_space.py

```diff
@@ -176,7 +176,7 @@ def test_design_space(self):
         self.assertEqual(len(ds.design_variables), 4)
         self.assertTrue(np.all(~ds.is_conditionally_acting))
 
-        ds.sample_valid_x(3)
+        ds.sample_valid_x(3, random_state=42)
```

Comment:
Make me think the new option `new_sampler` is not tested.

Reply:
Yes for EGO, butr it is a feature of `sample_valid_x` and to keep `DesignSpace` well-tested as it was, it would be nice to have here some specific tests of  `use_new_sampler`. Remember also we do not always run all EGO tests.